### ISSUE TITLE
[Python] Generated code does not respect nullable values

### ISSUE THREAD

**Issue by fridex** [swagger-api/swagger-codegen#8710]
##### Description

There is always generated following code which prevents from ``None`` (null) values:

```python
        if requirements_lock is None:
            raise ValueError("Invalid value for `<attr_name>`, must not be `None`")  # noqa: E501
```

Unfortunately this check is generated even if the swagger definition explicitly says the given model attribute is nullable (by providing ``nullable: true``).

##### Swagger-codegen version

The latest master to this date - master SHA 554d443a47cd172ed510b27935531a383b076189.

##### Command line used for generation

```console
java -jar swagger-codegen/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -i "${SWAGGER_YAML}" -l python
```
##### Steps to reproduce

Use any swagger definition and set ``nullable`` to true to any model attribute.

##### Related issues/PRs

None found.

##### Suggest a fix/enhancement

https://github.com/swagger-api/swagger-codegen/blob/554d443a47cd172ed510b27935531a383b076189/modules/swagger-codegen/src/main/resources/python/model.mustache#L100-L103

https://github.com/swagger-api/swagger-codegen/blob/554d443a47cd172ed510b27935531a383b076189/modules/swagger-codegen/src/main/resources/flaskConnexion/model.mustache#L119-L122

**Comment by heckad** [comment]
It is no fixed. When property with enum and nullable:true None doesn`t add in allowed_values.

**Comment by esseti** [comment]
I'm facing the same problem apparently. is this solved?

**Comment by anichari** [comment]
+1 any update on if there is going to be a code change so that the None check is added only if nullable is not True?

**Comment by HugoMario** [comment]
hey guys, sorry for delay on. this issue.

According to links pointing to code in templates, the check it's only added when the property is required. So, it does not make sense to have a nullable required property.

What do you guys think?

https://stackoverflow.com/questions/45575493/what-does-required-in-openapi-really-mean

**Comment by dRacz3** [comment]
@HugoMario 
Having the property listed in `required` would mean, that in the response the property must be there, but it does not specify it's value, which can be also `null` if `nullable = True` is set in the specification, therefore it is a bug imo.

Also, this issue seems to describe the same problem: #[10535](https://github.com/swagger-api/swagger-codegen/issues/10535)

(I have also experienced this issue with some project, and will not use the swagger codegen to generate my python client due to this problem). I have used the [docker images ](https://hub.docker.com/r/swaggerapi/swagger-codegen-cli-v3/tags?page=1&ordering=last_updated) for testing it. Checked back till a version 3.0.15, does not seem to work with any version :(
